docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -102,6 +102,289 @@ an exception, use the ``TrinoException`` class with the ``INVALID_ORDER_BY``
+
+            Map<ColumnHandle, Domain> supported = new HashMap<>();
+            Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+            for (int i = 0; i < columnHandles.size(); i++) {
+                JdbcColumnHandle column = columnHandles.get(i);
+                ColumnMapping mapping = columnMappings.get(i);
+                DomainPushdownResult pushdownResult = mapping.getPredicatePushdownController().apply(session, domains.get(column));
+                supported.put(column, pushdownResult.getPushedDown());
+                unsupported.put(column, pushdownResult.getRemainingFilter());
+            }
+
+            newDomain = TupleDomain.withColumnDomains(supported);
+            remainingFilter = TupleDomain.withColumnDomains(unsupported);
+
+            if (isComplexExpressionPushdown(session)) {
+                List<String> newExpressions = new ArrayList<>();
+                List<ConnectorExpression> remainingExpressions = new ArrayList<>();
+                for (ConnectorExpression expression : extractConjuncts(constraint.getExpression())) {
+                    Optional<String> converted = jdbcClient.convertPredicate(session, expression, constraint.getAssignments());
+                    if (converted.isPresent()) {
+                        newExpressions.add(converted.get());
+                    }
+                    else {
+                        remainingExpressions.add(expression);
+                    }
+                }
+                newConstraintExpressions = ImmutableSet.<String>builder()
+                        .addAll(handle.getConstraintExpressions())
+                        .addAll(newExpressions)
+                        .build().asList();
+                remainingExpression = Optional.of(and(remainingExpressions));
+            }
+            else {
+                newConstraintExpressions = ImmutableList.of();
+                remainingExpression = Optional.empty();
+            }
+        }
+
+        if (oldDomain.equals(newDomain) &&
+                handle.getConstraintExpressions().equals(newConstraintExpressions)) {
+            return Optional.empty();
+        }
+
+        handle = new JdbcTableHandle(
+                handle.getRelationHandle(),
+                newDomain,
+                newConstraintExpressions,
+                handle.getSortOrder(),
+                handle.getLimit(),
+                handle.getColumns(),
+                handle.getOtherReferencedTables(),
+                handle.getNextSyntheticColumnId());
+
+        return Optional.of(
+                remainingExpression.isPresent()
+                        ? new ConstraintApplicationResult<>(handle, remainingFilter, remainingExpression.get(), precalculateStatisticsForPushdown)
+                        : new ConstraintApplicationResult<>(handle, remainingFilter, precalculateStatisticsForPushdown));
+    }
+
+It splits the constraint expression into conjuncts and handles them
```

Comment:
What are conjuncts? We skip over `extractConjuncts` which might be important to understand how to process a constraint.